enginetest/enginetests.go

```diff
@@ -680,331 +521,275 @@ func TestAmbiguousColumnResolution(t *testing.T, harness Harness) {
+	TestQueryWithContext(t, ctx, e, `SELECT f.a, bar.b, f.b FROM foo f INNER JOIN bar ON f.a = bar.c order by 1`, expected, nil, nil)
 }
 
 func TestQueryErrors(t *testing.T, harness Harness) {
-	engine := NewEngine(t, harness)
-	defer engine.Close()
+	harness.SetSetup(MytableData, Pk_tablesData, MyhistorytableData)
+	for _, tt := range queries.ErrorQueries {
+		runQueryErrorTest(t, harness, tt)
+	}
+}
 
-	for _, tt := range errorQueries {
-		t.Run(tt.Query, func(t *testing.T) {
-			if sh, ok := harness.(SkippingHarness); ok {
-				if sh.SkipQueryTest(tt.Query) {
-					t.Skipf("skipping query %s", tt.Query)
-				}
-			}
-			AssertErrWithBindings(t, engine, harness, tt.Query, tt.Bindings, tt.ExpectedErr, tt.ExpectedErrStr)
-		})
+func MustQuery(ctx *sql.Context, e *sqle.Engine, q string) []sql.Row {
+	sch, iter, err := e.Query(ctx, q)
+	if err != nil {
+		panic(err)
 	}
+	rows, err := sql.RowIterToRows(ctx, sch, iter)
+	if err != nil {
+		panic(err)
+	}
+	return rows
 }
 
 func TestInsertInto(t *testing.T, harness Harness) {
-	for _, insertion := range InsertQueries {
-		e := NewEngine(t, harness)
-		defer e.Close()
-
-		TestQuery(t, harness, e, insertion.WriteQuery, insertion.ExpectedWriteResult, nil)
-		// If we skipped the insert, also skip the select
-		if sh, ok := harness.(SkippingHarness); ok {
-			if sh.SkipQueryTest(insertion.WriteQuery) {
-				t.Logf("Skipping query %s", insertion.SelectQuery)
-				continue
-			}
-		}
-		TestQuery(t, harness, e, insertion.SelectQuery, insertion.ExpectedSelect, nil)
+	harness.SetSetup(MydbData, MytableData, Mytable_del_idxData, KeylessData, NiltableData, TypestableData, EmptytableData, AutoincrementData, OthertableData)
+	for _, insertion := range queries.InsertQueries {
+		runWriteQueryTest(t, harness, insertion)
 	}
-	for _, script := range InsertScripts {
+
+	harness.SetSetup(MydbData)
+	for _, script := range queries.InsertScripts {
 		TestScript(t, harness, script)
 	}
 }
 
 func TestInsertIgnoreInto(t *testing.T, harness Harness) {
-	for _, script := range InsertIgnoreScripts {
+	harness.SetSetup(MydbData)
+	for _, script := range queries.InsertIgnoreScripts {
 		TestScript(t, harness, script)
 	}
 }
 
 func TestInsertIntoErrors(t *testing.T, harness Harness) {
-	for _, expectedFailure := range InsertErrorTests {
-		t.Run(expectedFailure.Name, func(t *testing.T) {
-			if sh, ok := harness.(SkippingHarness); ok {
-				if sh.SkipQueryTest(expectedFailure.Query) {
-					t.Skipf("skipping query %s", expectedFailure.Query)
-				}
-			}
-			AssertErr(t, NewEngine(t, harness), harness, expectedFailure.Query, nil)
-		})
+	harness.SetSetup(mytable...)
+	for _, expectedFailure := range queries.InsertErrorTests {
+		runGenericErrorTest(t, harness, expectedFailure)
 	}
-	for _, script := range InsertErrorScripts {
+
+	harness.SetSetup(MydbData)
+	for _, script := range queries.InsertErrorScripts {
 		TestScript(t, harness, script)
 	}
 }
 
 func TestBrokenInsertScripts(t *testing.T, harness Harness) {
 	t.Skip()
```

Comment:
Fix this: we want a skipped tests for each broken query, not just a single skippe dtest